Ginger/Ginger/ApplicationModelsLib/APIModels/APIModelWizard/ScanAPIModelWizardPage.xaml.cs

```diff
@@ -284,7 +394,101 @@ public class SoapData
 
         private void IsSelected_FieldSelection_Click(object sender, RoutedEventArgs e)
         {
-            ObservableList<ApplicationAPIModel> CheckedList = GingerCore.General.ConvertListToObservableList(AddAPIModelWizard.AAMList.Where(x => x.IsSelected == true).ToList());
-        }        
+            ObservableList<ApplicationAPIModel> CheckedList = GingerCore.General.ConvertListToObservableList(AddAPIModelWizard.LearnedAPIModelsList.Where(x => x.IsSelected == true).ToList());
+        }
+
+        private void xCompareAndMergeButton_Click(object sender, RoutedEventArgs e)
+        {
+            ShowMergerPage(sender);
+        }
+
+        void ShowMergerPage(object sender)
+        {
+            DeltaAPIModel deltaAPI = null;
+            var fEl = sender as FrameworkElement;
+            if (fEl != null)
+                deltaAPI = fEl.DataContext as DeltaAPIModel;
+            if (deltaAPI != null)
+            {
+                if (deltaAPI.mergerPageObject == null)
+                {
+                    deltaAPI.mergerPageObject = new MergerPage(deltaAPI, (Window)AddAPIModelWizard.mWizardWindow);
+                }
+
+                (deltaAPI.mergerPageObject as MergerPage).ShowAsWindow();
+            }
+        }
+
+        private void XHandlingOperation_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            ComboBox handlingOpCB = sender as ComboBox;
+            string mergerDescription = Convert.ToString(DeltaAPIModel.GetEnumDescription(DeltaAPIModel.eHandlingOperations.MergeChanges));
+
+            if (handlingOpCB != null && handlingOpCB.SelectedItem != null)
+            {
+                DeltaAPIModel.eHandlingOperations selectedOperation = DeltaAPIModel.GetValueFromDescription<DeltaAPIModel.eHandlingOperations>(Convert.ToString(handlingOpCB.SelectedItem));
+
+                DeltaAPIModel deltaAPI = null;
+                var fEl = sender as FrameworkElement;
+                if (fEl != null)
+                    deltaAPI = fEl.DataContext as DeltaAPIModel;
+                if (deltaAPI != null)
+                {
+                    deltaAPI.defaultOperation = Convert.ToString(handlingOpCB.SelectedItem);
```

Comment:
still i'm not getting why those enum-string playing are needed, why binding not been used here? why need to cast from enum to string or vis versa?

Reply:
coz we're showing the enum descriptions in the combo box items and when combo box selection is changed, we need to update default operation and thus mapping is required here.